core/amber/src/main/scala/edu/uci/ics/texera/web/service/ExecutionResultService.scala

```diff
@@ -225,18 +233,15 @@ class ExecutionResultService(
             case (opId, info) =>
               val oldInfo = oldState.resultInfo.getOrElse(opId, OperatorResultMetadata())
               buf(opId.id) = ExecutionResultService.convertWebResultUpdate(
-                sinkOperators(opId),
+                workflowIdentity,
+                physicalPlan.getPhysicalOpsOfLogicalOp(opId),
                 oldInfo.tupleCount,
                 info.tupleCount
               )
-              if (
-                StorageConfig.resultStorageMode.toLowerCase == "mongodb"
-                && !opId.id.startsWith("sink")
-              ) {
-                val sinkOp = sinkOperators(opId)
+              if (StorageConfig.resultStorageMode.toLowerCase == "mongodb") {
                 val opStorage = ResultStorage
-                  .getOpResultStorage(sinkOp.getContext.workflowId)
-                  .get(sinkOp.getUpstreamId.get)
+                  .getOpResultStorage(workflowIdentity)
+                  .get(physicalPlan.getPhysicalOpsOfLogicalOp(opId).head.id.logicalOpId)
```

Comment:
Is`physicalPlan.getPhysicalOpsOfLogicalOp(opId).head.id.logicalOpId` == `opId`?

Reply:
yes XD. fixed.